Cpp_test/lib/ipc-pipe-receive.cpp

```diff
@@ -0,0 +1,42 @@
+#include "ipc-pipe-receive.h"
+
+IPCPipeReceive::~IPCPipeReceive()
+{
+	close(this->pd);
+	unlink(this->info.method_name);
+}
+
+void IPCPipeReceive::init()
+{
+	this->file_handler.setup_file(info.file_name, FileMode::WRITE);
+	std::cout << "Waiting for sender... " << std::endl;
+	this->pd = open(this->info.method_name, O_RDONLY | O_NONBLOCK);
```

Comment:
If you use the O_NONBLOCK flag you should take care about race issues (like the receiver is faster than the sender)

Reply:
For instance, we had to launch the sender before the receiver.

Reply:
Yes, I will use do while here and set it to sleep a few seconds to wait for sender, so sender always launches before receiver.